src/SettingsMigrator/Migrator.cs

```diff
@@ -153,65 +152,65 @@ private static void MigrateTestSettings(string oldTestSettingsPath, string newRu
         {
-            testTimeout = testSettingsNodes.Timeout.Attributes[TestTimeoutAttributeName].Value;
+            testTimeout = testSettingsNodes?.Timeout?.Attributes?[TestTimeoutAttributeName]?.Value;
         }
 
         string? runTimeout = null;
-        if (testSettingsNodes.Timeout != null && testSettingsNodes.Timeout.Attributes[RunTimeoutAttributeName] != null)
+        if (testSettingsNodes?.Timeout?.Attributes?[RunTimeoutAttributeName] != null)
         {
-            runTimeout = testSettingsNodes.Timeout.Attributes[RunTimeoutAttributeName].Value;
+            runTimeout = testSettingsNodes?.Timeout?.Attributes?[RunTimeoutAttributeName]?.Value;
         }
 
         string? parallelTestCount = null;
-        if (testSettingsNodes.Execution != null && testSettingsNodes.Execution.Attributes[ParallelTestCountAttributeName] != null)
+        if (testSettingsNodes?.Execution?.Attributes?[ParallelTestCountAttributeName] != null)
         {
-            parallelTestCount = testSettingsNodes.Execution.Attributes[ParallelTestCountAttributeName].Value;
+            parallelTestCount = testSettingsNodes?.Execution?.Attributes?[ParallelTestCountAttributeName]?.Value;
         }
 
         string? hostProcessPlatform = null;
-        if (testSettingsNodes.Execution != null && testSettingsNodes.Execution.Attributes[HostProcessPlatformAttributeName] != null)
+        if (testSettingsNodes?.Execution?.Attributes?[HostProcessPlatformAttributeName] != null)
         {
-            hostProcessPlatform = testSettingsNodes.Execution.Attributes[HostProcessPlatformAttributeName].Value;
+            hostProcessPlatform = testSettingsNodes?.Execution?.Attributes?[HostProcessPlatformAttributeName]?.Value;
         }
 
         // WebTestRunConfiguration node.
-        if (testSettingsNodes.WebSettings != null)
+        if (testSettingsNodes?.WebSettings != null)
         {
-            runSettingsXmlDoc.DocumentElement.AppendChild(runSettingsXmlDoc.ImportNode(testSettingsNodes.WebSettings, deep: true));
+            runSettingsXmlDoc?.DocumentElement?.AppendChild(runSettingsXmlDoc.ImportNode(testSettingsNodes.WebSettings, deep: true));
         }
 
         // LegacySettings node.
-        AddLegacyNodes(testSettingsNodes, testTimeout, parallelTestCount, hostProcessPlatform, runSettingsXmlDoc);
+        AddLegacyNodes(testSettingsNodes!, testTimeout, parallelTestCount, hostProcessPlatform, runSettingsXmlDoc!);
 
         // TestSessionTimeout node.
         if (!runTimeout.IsNullOrEmpty())
         {
-            AddRunTimeoutNode(runTimeout, runSettingsXmlDoc);
+            AddRunTimeoutNode(runTimeout, runSettingsXmlDoc!);
         }
 
         // DataCollectors node.
-        if (testSettingsNodes.Datacollectors != null && testSettingsNodes.Datacollectors.Count > 0)
+        if (testSettingsNodes?.Datacollectors != null && testSettingsNodes.Datacollectors.Count > 0)
```

Comment:
According to nullability above, `testSettingsNodes`  cannot be null

```suggestion
        if (testSettingsNodes.Datacollectors != null && testSettingsNodes.Datacollectors.Count > 0)
```

Reply:
I believe you can also simplify further

```suggestion
        if (testSettingsNodes.Datacollectors?.Count > 0)
```